gdal/frmts/mrf/mrf_band.cpp

```diff
@@ -186,6 +251,99 @@ static void *DeflateBlock(buf_mgr &src, size_t extrasize, int flags) {
     return src.buffer;
 }
 
+#if defined(ZSTD_SUPPORT)
+
+// Arange bytes by rank in the input
+static void rankfilter(buf_mgr& src, size_t factor) {
+    if (factor > 1) {
+        std::vector<char> tempb(src.size);
+        char* d = tempb.data();
+        for (size_t j = 0; j < factor; j++)
+            for (size_t i = j; i < src.size; i += factor)
+                *d++ = src.buffer[i];
+        memcpy(src.buffer, tempb.data(), src.size);
+    }
+    // Apply per byte delta
+    auto guard = src.buffer + src.size;
+    char b(0);
+    for (auto p = src.buffer; p < guard; p++) {
+        char temp = *p;
+        *p -= b;
```

Comment:
This will have wrap-around effect, which is undefined behavior on a signed type. See https://github.com/OSGeo/gdal/blob/master/gdal/frmts/rmf/rmfdem.cpp#L98 for a safer way of doing that (to be adapted to the uint8 case)